horace_core/utilities/split_vector_fixed_sum.m

```diff
@@ -0,0 +1,113 @@
+% cumulative_sum   The cumulative sum of 'numeric_vector'. (optional)
+%
+% Output:
+% -------
+% chunks   Cell array of vectors.
+% idxs     The indices at which the input vector was "split". Has size
+%          [2, n], where n is numel(chunks). Each idxs(:, chunk_num) is the
+%          lower and upper index into 'numeric_vector' of chunks{chunk_num}.
+%
+% Example:
+% --------
+% >> numeric_vector = [3, 2, 6, 0, 5, 3, 1, 0, 1, 24, 4, 2, 3, 0];
+% >> chunk_sum = 10;
+% >> [chunks, idxs] = split_vector_max_sum(numeric_vector, chunk_sum)
+%   chunks =
+%       { [3, 2, 5], [1, 0, 5, 3, 1, 0], [1, 9], [10], [5, 4, 1], [1, 3, 0] }
+%   idxs =
+%       1     4     9    10    10    12
+%       4     8    10    10    12    14
+%
+% >> numeric_vector = [24];
+% >> chunk_sum = 10;
+%   chunks =
+%       { [10], [10], [4] }
+%   idxs =
+%       1
+%       1
+%
+if isempty(numeric_vector)
+    chunks = {};
+    idxs = zeros(2, 0);
+    return;
+end
+
+validateattributes(numeric_vector, {'numeric'}, {'vector', 'nonnegative'});
+validateattributes(chunk_sum, {'numeric'}, {'scalar', 'positive'});
+
+if ~exist('cumulative_sum', 'var')
+    cumulative_sum = cumsum(numeric_vector);
+end
+vector_sum = cumulative_sum(end);
+num_chunks = ceil(vector_sum/chunk_sum);
+
+chunks = cell(1, num_chunks);
+idxs = zeros(2, num_chunks);
+end_idx = 1;
+allocated = 0;
+for chunk_num = 1:num_chunks
+    remaining_sum = vector_sum - allocated;
+    chunk_sum = min(chunk_sum, remaining_sum);
+
+    if chunk_num == 1
+        % For the first iteration, we need >= to catch leading zeros
+        % This index may contribute its full value, or only part of its value
+        start_idx = (end_idx - 1) + find(cumulative_sum(end_idx:end) >= 0, 1);
+    else
+        % We ignore leading zeros on subsequent iterations as they will have
+        % been assigned to the previous chunk
+        start_idx = (end_idx - 1) + find(cumulative_sum(end_idx:end) > 0, 1);
+    end
+
+    % First value in the current chunk
+    leftover_begin = cumulative_sum(start_idx);
+
+    % Offset cumulative sum so next positive value corresponds to index of
+    % the final value of current chunk
+    cumulative_sum = cumulative_sum - chunk_sum;
+    end_idx = (start_idx - 1) + find(cumulative_sum(start_idx:end) > 0, 1);
+
+    if isempty(end_idx)
+        % No end index found, so we must be at the end of the vector
+        end_idx = numel(numeric_vector);
+    end
+
+    if start_idx == end_idx
+        % All vector values in single value for this chunk
+        if ~exist('leftover_end', 'var')
+            leftover_end = 0;
+        end
+        chunks{chunk_num} = min(chunk_sum, numeric_vector(start_idx) - leftover_end);
+    else
+        % Add an empty element to the end, which will be used to hold a part of
+        % the end index's value. If there's no remainder on the end index's
+        % value, the empty element will be removed
```

Comment:
I am not clear why reshape is required. A quick experiment at the comment line seems to say that [begin, nv[start+1,end-1],0] produces a 1xn vector fine

Reply:
The `reshape` is important, otherwise the function will fail if the input is a column vector.
```matlab
>> row_vec = [1, 2, 3, 4];
>> col_vec = row_vec';
>> size(row_vec(1:end - 1))
ans =

     1     3

size(col_vec(1:end - 1))
ans =

     3     1
```

Reply:
Another (probably cleaner) solution is to reshape the whole vector at the start of the function. I'll do that instead